packages/terra-slide-panel/src/SlidePanel.jsx

```diff
@@ -85,33 +85,65 @@ const defaultProps = {
   panelSize: 'small',
 };
 
+/**
+ * This method ensures that aria-expanded attribute is allowed with the node role to avoid accessibility violations.
+ * More info on associated roles can be found here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#associated_roles
+ * @param {HTMLElement} node - a node to check
+ * @returns true if node's role is compatible with aria-expanded attribute
+ */
+const roleIsCompartible = (node) => {
+  const role = node.getAttribute('role');
+  return (
+    role === 'application'
+    || role === 'button'
+    || role === 'checkbox'
+    || role === 'combobox'
+    || role === 'gridcell'
+    || role === 'link'
+    || role === 'listbox'
+    || role === 'menuitem'
+    || role === 'row'
+    || role === 'rowheader'
+    || role === 'tab'
+    || role === 'treeitem'
+    || role === null
+    || role === undefined
+  );
+};
+
 class SlidePanel extends React.Component {
   constructor(props) {
     super(props);
     this.setPanelNode = this.setPanelNode.bind(this);
     this.mainNode = React.createRef();
-    this.setLastClicked = this.setLastClicked.bind(this);
     this.setDisclosingNode = this.setDisclosingNode.bind(this);
     this.mainAriaDescribedByID = `detail-panel-warning-${uuidv4()}`;
   }
 
   componentDidUpdate(prevProps) {
     if (this.props.isOpen && this.props.isOpen !== prevProps.isOpen) {
       // Save the disclosing node for returning focus when panel is closed
-      this.setDisclosingNode(this.lastClicked);
+      this.setDisclosingNode(document.activeElement);
       this.panelNode.focus();
       return;
     }
 
     if (!this.props.isOpen && this.props.isOpen !== prevProps.isOpen) {
       if (this.disclosingNode?.focus) {
         // Return focus to the disclosing element
-        this.disclosingNode.setAttribute('aria-expanded', 'false');
+        if (this.disclosingNode.getAttribute('aria-expanded')) {
+          // Set aria-expanded attribute to false for nodes that had that attribute only
+          this.disclosingNode.setAttribute('aria-expanded', 'false');
```

Comment:
I am not clear on why we need to set the aria-expanded attribute on the disclosing node.  That element is not being expanded.

Reply:
aria-expanded attribute was not added at this change, it was added before. I'll check with @mjpalazzo if it can be removed.